lib/elastic_whenever/option.rb

```diff
@@ -128,15 +128,13 @@ def initialize(args)
         end
       end.parse(args)
 
-      @credentials = if profile
-                       Aws::SharedCredentials.new(profile_name: profile)
-                     elsif access_key && secret_key
+      @credentials = if access_key && secret_key
                        Aws::Credentials.new(access_key, secret_key)
                      end
     end
 
     def aws_config
-      @aws_config ||= { credentials: credentials, region: region }.delete_if { |_k, v| v.nil? }
+      @aws_config ||= { credentials: credentials, region: region, profile: @credentials ? nil : profile }.delete_if { |_k, v| v.nil? }
```

Comment:
Do we need to check for `@credentials` here? I'm thinking it might be a good idea to simply pass the `profile`.

```suggestion
      @aws_config ||= { credentials: credentials, region: region, profile: profile }.delete_if { |_k, v| v.nil? }
```

Reply:
I think passing it would be redundant since there's no need for looking up the profile and I wouldn't even know what would happen when someone would pass profile _and_ credentials 🤔

Reply:
I agree that there is no valid case for passing a profile with static credentials, but I also don't think there is a need to intentionally ignore a profile passed with `--profile`.
I believe that it is better to reflect the flags specified by users as is, to keep the implementation simple and intuitive.